docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,252 @@
+paid from the maintenance pool.
+
+Just like in the case of relay entry submission, the order in which operators
+are supposed to submit DKG result is enforced on-chain and operators who missed
+their round are removed from the sortition pool for at least 2 weeks and do
+not earn rewards for that time. The first member eligible to submit the DKG result
+is a member with index `hash(new_group_pubkey) % group_size`, then also 
+`(hash(new_group_pubkey) % group_size) + 1`. For example, if 
+`hash(new_group_pubkey) % group_size = 62`, `group_size = 64`, and member `9` 
+submitted DKG result, members with indexes `{62, 63, 64, 1, 2, 3, 4, 5, 6, 7, 8}` 
+should be removed from the sortition pool in the same transaction in which 
+member `9` submitted DKG result.
+
+The default time for each member to submit DKG result should be set to 10 blocks
+and should become a governable parameter.
+
+To make this scheme work, both the relay entry and the DKG result submission
+transactions need to have predictable gas costs. Although it is guaranteed for
+the latter, the former, in v1 of the beacon, depends on the gas cost of
+executing a callback and needs to be optimized - more on that in the
+ <<callbacks,Callbacks>> section.
+
+T rewards will be distributed continuously to all operators in the beacon
+sortition pool, just like in the case of tBTC v2.
+
+[[callbacks]]
+=== Callbacks
+
+In v1 of the random beacon, a callback is executed in the same transaction in
+which relay entry is submitted with a gas limit of 2M. This allows applications
+using the random beacon to avoid an additional complexity at their end but comes
+at the cost of random beacon operators who need to have enough ETH on their
+balance. This approach is not compatible with the idea that all rewards will be
+solely in T given that it is impossible to establish the cost of executing a
+callback in T.
+
+The fact the full - even the most complex - callback is executed in the same
+transaction in which the relay entry is submitted gives an impression of better
+security. This impression is false though, given that the entry to be submitted
+is visible in the mempool and smart attackers can have their transactions mined
+faster to, for example, put the sortition pool using the entry for group
+selection in the desired state. This issue needs to be solved on the sortition
+pool side with initiation time for new operators in the pool and/or state lock.
+
+Instead of executing any callback in the same transaction in which relay entry
+is submitted, we should allow only simple, low gas budget callbacks with a gas
+limit controlled by the governance. Applications wanting to use a relay entry
+should submit another transaction using the relay entry value previously set by
+the random beacon. They should also employ an additional security check making
+sure the entry submitted is only valid for a certain number of blocks to avoid
```

Comment:
Is that `certain number of blocks` a governable parameter?

Reply:
Not for the random beacon. The random beacon does not care if and how quickly the application will use the relay entry the beacon produced. It is up to the application to have the checks needed.